backbone.js

```diff
@@ -543,6 +543,18 @@
       return this.set(attr, void 0, _.extend({}, options, {unset: true}));
     },
 
+    // accepts an attribute name (or array of attribute names) and a transformer
+    // function which maps the current attribute value(s) to new attribute value(s)
+    update: function(attrName, transformer, setOptions) {
+      var attrNames = _.isArray(attrName) ? attrName : [attrName];
+      var newValues = _.reduce(attrNames, function(valueCollector, attr) {
+        valueCollector[attr] = transformer.call(this, this.get(attr));
```

Comment:
@erosb ok.  That does limit the ability to use es6 arrow functions though, since that won't work with them (this will be in the callers scope and it will ignore the `call` binding.  Passing it as an argument is more flexible and obvious.  You could also `this` bind the model and just pass the attributes hash as the 3rd argument, similar to [Backbone.Validation](https://github.com/thedersen/backbone.validation)